eos/form-factors/analytic-b-to-gamma-qcdf.cc

```diff
@@ -0,0 +1,187 @@
+/* vim: set sw=4 sts=4 et foldmethod=syntax : */
+
+/*
+ * Copyright (c) 2022 Danny van Dyk
+ * Copyright (c) 2022 Philip Lüghausen
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/form-factors/analytic-b-to-gamma-qcdf.hh>
+#include <eos/utils/exception.hh>
+
+#include <numeric>
+
+namespace eos
+{
+    AnalyticFormFactorBToGammaQCDF::AnalyticFormFactorBToGammaQCDF(const Parameters & p, const Options & o):
+        omega_0(   p[par_qname("omega_0")],    *this),
+        blcdas(BMesonLCDAs::make("flvd2022", p, o))
+    {
+        // Verify once that BMesonLCDAs is compatible with this implementation
+        Weights weights;
+        auto [phi_plus_coeff, phi_plus_coeff_end] = blcdas->coefficient_range(1.0);
+
+        if (weights.size() - std::distance(phi_plus_coeff, phi_plus_coeff_end) > 0)
+        {
+            throw InternalError("The number of weights implemented is smaller than the number of coefficients of phi_+");
+        }
+    }
+
+    std::string
+    AnalyticFormFactorBToGammaQCDF::par_qname(const std::string & _name)
+    {
+        qnp::Name name(_name);
+        return QualifiedName(qnp::Prefix("B_u"), name, qnp::Suffix("FLvD2022")).str();
+    }
+
+    FormFactors<PToGamma> *
+    AnalyticFormFactorBToGammaQCDF::make(const Parameters & p, const Options & o)
+    {
+        return new AnalyticFormFactorBToGammaQCDF(p, o);
+    }
+
+    double
+    AnalyticFormFactorBToGammaQCDF::L0() const
+    {
+        Weights c = {1., 0, 1./3., 0, 1./5., 0, 1./7., 0, 1./9.};
```

Comment:
Coding style:
```c++
c = { 1.0, ..., 1.0 / 9.0 };
```
 - don't drop the ``0`` after the point.
 - leave a space to the left and right of the operator.
 - leave a space to the right of ``{`` and to the left of ``}``, for legibility.

Reply:
Okay! All other weights are generated via Mathematicas `CForm` though, which I would simply keep. These long expressions are not pretty either way.
Maybe it would be sensible to hide them away in a separate implementation header.